src/hooks/routing/useRouterTrade.ts

```diff
@@ -82,7 +82,7 @@ export function useRouterTrade(
   return useMemo(() => {
     if (!amountSpecified || isError || queryArgs === skipToken) {
       return TRADE_INVALID
-    } else if (data === NO_ROUTE) {
+    } else if (data === NO_ROUTE && isValid) {
```

Comment:
`isValid` represents whether the cached result (`data`) is the actual result for the current `queryArgs` if it's not, then we're fetching a new result for the most relevant args.

Reply:
helpful comment, ty!